service/src/main/java/bio/terra/profile/service/azure/ApplicationService.java

```diff
@@ -16,12 +18,53 @@ public ApplicationService(CrlService crlService) {
     this.crlService = crlService;
   }
 
+  /**
+   * Retrieves the managed applications for a given Azure subscription
+   *
+   * @param subscriptionId Azure subscription ID to be queried
+   * @return List of managed applications present in the subscription
+   */
   public Stream<Application> getApplicationsForSubscription(UUID subscriptionId) {
     var appMgr = crlService.getApplicationManager(subscriptionId);
     return appMgr.applications().list().stream();
   }
 
+  /**
+   * Retrieves the tenant associated with a given Azure subscription ID
+   *
+   * @param subscriptionId Azure subscription ID to be queried
+   * @return UUID for the associated tenant
+   */
   public UUID getTenantForSubscription(UUID subscriptionId) {
     return crlService.getTenantForSubscription(subscriptionId);
   }
+
+  /**
+   * Adds a tag to the MRG associated with a given profile.
+   *
+   * @param tenantId Tenant ID associated with the MRG
+   * @param subscriptionId Subscription ID associated with the MRG
+   * @param managedResourceGroupId ID for the MRG
+   * @param tag Tag name
+   * @param value Tag value
+   * @throws DuplicateTagException if the tag is already present.
+   */
+  public void addTagToMrg(
+      UUID tenantId, UUID subscriptionId, String managedResourceGroupId, String tag, String value) {
```

Comment:
Why is `throws DuplicateTagException` not part of the function declaration explicitly?

Reply:
I typically only add a `throws` for checked exceptions, not for runtime exceptions.